File: src/data/courses/dsa/dynamic-programming/tile-counting/ways-reach-nth-stair/code/python/ways_reach_nth_stair.py
class Solution:
    def waysToReach(self, n):
        if n <= 1: return 1
        if n == 2: return 2
        a, b, c = 1, 1, 2
        for _ in range(3, n+1): a, b, c = b, c, a+b+c
        return c

File: src/data/courses/dsa/dynamic-programming/tile-counting/ways-reach-nth-stair/code/java/WaysReachNthStair.java
class Solution {
    public int waysToReach(int n) {
        if (n<=1) return 1; if (n==2) return 2;
        int a=1,b=1,c=2;
        for (int i=3;i<=n;i++){int d=a+b+c;a=b;b=c;c=d;} return c;
    }
}